src/components/views/settings/JoinRuleSettings.tsx

```diff
@@ -49,13 +50,20 @@ interface IProps {
 const JoinRuleSettings = ({ room, promptUpgrade, aliasWarning, onError, beforeChange, closeSettingsFn }: IProps) => {
     const cli = room.client;
 
+    const roomSupportsKnocking = doesRoomVersionSupport(room.getVersion(), PreferredRoomVersions.KnockingRooms);
+    const preferredKnockingVersion = !roomSupportsKnocking && promptUpgrade
```

Comment:
Is this unused?

Reply:
Yeah going to use it to check the room version supports knocking or not